core/unit_test/view/TestViewCustomizationAllocationType.hpp

```diff
@@ -374,3 +374,48 @@ void test_deep_copy_single_element_view() {
 TEST(TEST_CATEGORY, view_customization_deep_copy_single_element_view) {
   test_deep_copy_single_element_view();
 }
+
+template <class ViewT, class FirstArg, std::integral... Sizes>
+void test_required_span_size_single_rank(size_t expected_size,
+                                         FirstArg first_arg, Sizes... args) {
+  ViewT view(first_arg, args...);
+  // Lets get the required size two ways: based on mapping + accessor and using
+  // the required_allocation_size
+  size_t extra_dim = view.accessor().size;
+  size_t span_size = view.mapping().required_span_size();
+  size_t span_size_based_bytes =
+      span_size * sizeof(std::remove_pointer_t<decltype(view.data())>) *
```

Comment:
Remind me in what circumstance `sizeof(std::remove_pointer_t<decltype(view.data())` can differ from `sizeof(ViewT::data_type)`

Reply:
The above is also the reason for having the private member `view::raw_allocation_value_type` I guess.